dask/dataframe/io/parquet/utils.py

```diff
@@ -10,6 +13,86 @@
 class Engine:
     """The API necessary to provide a new Parquet reader/writer"""
 
+    @classmethod
+    def extract_filesystem(
+        cls,
+        urlpath,
+        filesystem,
+        dataset_options,
+        open_file_options,
+        storage_options,
+    ):
+        """Extract filesystem object from urlpath or user arguments
+
+        This classmethod should only be overridden for engines that need
+        to handle filesystem implementations other than ``fsspec``
+        (e.g. ``pyarrow.fs.S3FileSystem``).
+
+        Parameters
+        ----------
+        urlpath: str or List[str]
+            Source directory for data, or path(s) to individual parquet files.
+        filesystem: "fsspec" or fsspec.AbstractFileSystem
+            Filesystem backend to use. Default is "fsspec"
+        dataset_options: dict
+            Engine-specific dataset options.
+        open_file_options: dict
+            Options to be used for file-opening at read time.
+        storage_options: dict
+            Options to be passed on to the file-system backend.
+
+        Returns
+        -------
+        fs: Any
+            A global filesystem object to be used for metadata
+            processing and file-opening by the engine.
+        paths: List[str]
+            List of data-source paths.
+        dataset_options: dict
+            Engine-specific dataset options.
+        open_file_options: dict
+            Options to be used for file-opening at read time.
+        """
+
+        # Check if fs was specified as a dataset option
+        fs = dataset_options.pop("fs", "fsspec")
+        if filesystem is not None:
+            fs = filesystem
```

Comment:
It looks like there are now multiple possible ways one can specify a filesystem (i.e. the `filesystem=` kwarg and as a `dataset_options`). We should probably raise if they're both specified, especially if they conflict in some way. 

Side note: Should we deprecate the `dataset_options` path for specifying a filesystem in favor of the top-level `filesystem=` kwarg?

Reply:
>We should probably raise if they're both specified

Sure - I agree that we should raise an error if both are defined.

>Should we deprecate the dataset_options path for specifying a filesystem in favor of the top-level filesystem= kwarg?

I'd prefer to support the case that the user passes in `pyarrow.dataset` options under the `dataset` argument (unless you feel strongly).

Reply:
Sure, let's just raise an error when they're both defined then